validator/rpc/accounts.go

```diff
@@ -148,3 +149,28 @@ func (s *Server) BackupAccounts(
 		ZipFile: buf.Bytes(),
 	}, nil
 }
+
+// DeleteAccounts deletes accounts from a user if their wallet is an imported wallet.
+func (s *Server) DeleteAccounts(
+	ctx context.Context, req *pb.DeleteAccountsRequest,
+) (*pb.DeleteAccountsResponse, error) {
+	if req.PublicKeysToDelete == nil || len(req.PublicKeysToDelete) < 1 {
+		return nil, status.Error(codes.InvalidArgument, "No public keys specified to delete")
+	}
+	if s.wallet == nil || s.keymanager == nil {
+		return nil, status.Error(codes.FailedPrecondition, "No wallet found")
+	}
+	if s.wallet.KeymanagerKind() != keymanager.Imported {
```

Comment:
We have since changed the requirements and now both imported and derived keymanagers can delete keys. Remote cannot, so we should update here

Reply:
How would one test a derived wallet delete?   accounts.DeleteAccounts supports both imported and derived but only tests the imported wallets (check [here](https://github.com/prysmaticlabs/prysm/blob/ce53b7e5f515a0ec380b30468acd8db614ccaac5/validator/accounts/accounts_delete_test.go#L19)
If i get it to work in rpc , we can update the accounts module as well.

Reply:
A derived wallet is internally implemented as an imported wallet, that's why testing an imported wallet is enough for the most part.

An example of setting up a derived wallet for testing can be found here:
https://github.com/prysmaticlabs/prysm/blob/28e4a3b7e818bfa61ad7e4a8a8169d23d1dfed49/validator/client/wait_for_activation_test.go#L311